src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -278,6 +282,324 @@ struct mac_addr_list
+      idx++;
+      buffer = {0};
+      ifs.getline(buffer.data(), sz);
+    }
+
+    if (idx < XLAPC_STATUS_PER_SLOT)
+      std::cout << "\nINFO: Incomplete LAPC data in " << path << std::endl;
+
+    return val_buf;
+
+  }
+};
+
+
+/* SPC status 
+ * In PCIe Linux, access the sysfs file for SPC to retrieve the SPC status
+ */ 
+struct spc_status
+{
+  using result_type = query::spc_status::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key, const boost::any& arg1)
+  {
+    const auto dbg_ip_data = boost::any_cast<query::spc_status::debug_ip_data_type>(arg1);
+    auto pdev = get_pcidev(device);
+
+    std::string spc_name("spc_");
+    spc_name = spc_name + std::to_string(dbg_ip_data->m_base_address);
+
+    std::string name_path = pdev->get_sysfs_path(spc_name.c_str(), "name");
+
+    std::size_t pos = name_path.find_last_of('/');
+    std::string path = name_path.substr(0, pos+1);
+    path += "status";
+
+    std::vector<uint32_t> val_buf(XLAPC_STATUS_PER_SLOT, 0);
+
+    std::ifstream ifs(path);
+    if (!ifs) {
+      std::cout << "\nINFO: Incomplete SPC data in " << path << std::endl;
+      return val_buf;
+    }
+
+    constexpr size_t sz = 256;
+    std::array<char, sz> buffer = {0};
+    ifs.getline(buffer.data(), sz);
+
+    size_t idx = 0;
+    while(!ifs.eof()) {
+      val_buf[idx] = std::stoi((const char*)(&buffer), nullptr, 10);
+      idx++;
+      buffer = {0};
+      ifs.getline(buffer.data(), sz);
+    }
+
+    if (idx < XLAPC_STATUS_PER_SLOT)
+      std::cout << "\nINFO: Incomplete SPC data in " << path << std::endl;
+
+    return val_buf;
+  }
+};
+
+
+/* Accelerator Deadlock Detector status 
+ * In PCIe Linux, access the sysfs file for Accelerator Deadlock Detector to retrieve the deadlock status
+ */ 
+struct accel_deadlock_status
+{
+  using result_type = query::accel_deadlock_status::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key, const boost::any& arg1)
+  {
+    const auto dbg_ip_data = boost::any_cast<query::accel_deadlock_status::debug_ip_data_type>(arg1);
+    auto pdev = get_pcidev(device);
+
+    result_type val_buf = 0;
+
+    // read counter values
+    std::string mon_name("accel_deadlock_");
+    mon_name = mon_name + std::to_string(dbg_ip_data->m_base_address);
+
+    std::string name_path = pdev->get_sysfs_path(mon_name.c_str(), "name");
+
+    std::size_t pos = name_path.find_last_of('/');
+    std::string path = name_path.substr(0, pos+1);
+    path += "status";
+
+    std::ifstream ifs(path);
+    if (!ifs) {
+      return val_buf;
+    }
+
+    constexpr size_t sz = 256;
+    std::array<char, sz> buffer = {0};
+    ifs.getline(buffer.data(), sz);
+
+    if (!ifs.eof()) {
+      val_buf = std::stoi((const char*)(&buffer), nullptr, 10);
+    } else {
+      std::cout << "\nINFO: Incomplete Accelerator Deadlock detector status in " << path << std::endl;
+    }
+
+    return val_buf;
+
```

Comment:
Extra slot should be removed.

Reply:
I could not follow which one are you referring as extra slot.

Reply:
@rozumx Perhaps that you are parsing a full `std::vector` of data while you are using only the first element?

Reply:
@keryell , Stephen had added the comment before "get_counter_status_from_sysfs" method was introduced. So, he is definitely not referring to using std::vector for a single element.

Reply:
@rozumx Could you please explain this comment? I could not follow what are you referring as extra slot.